sgf/scanner.go

```diff
@@ -0,0 +1,160 @@
+package sgf
+
+import (
+	"bufio"
+	"bytes"
+	"io"
+)
+
+const (
+	Eof = iota
+	Whitespace
+	LeftParen
+	RightParen
+	LeftBracket
+	RightBracket
+	Semicolon
+	Backslash
+	String
+)
+
+var eof = rune(0)
+
+// isWhiteSpace is a check to see if a rune is whitespace (space, tab, newline, carriage return)
+func isWhitespace(ch rune) bool {
+	return ch == ' ' || ch == '\t' || ch == '\n' || ch == '\r'
+}
+
+// isSpecial is a check for special characters used in SGF files
+func isSpecial(ch rune) bool {
+	return ch == '(' || ch == ')' || ch == '[' || ch == ']' || ch == ';'
+}
+
+// The scanner will "tokenize" the input into Token structs
+type Token struct {
+	Type int
+	Raw  string
+}
+
+// Scanner wraps bufio.Reader but has the public Scan() function to scan tokens
+type Scanner struct {
+	*bufio.Reader
+}
+
+// Creates a new scanner object out of an io.Reader interface
+func NewScanner(r io.Reader) *Scanner {
```

Comment:
It's probably helpful to have a method 

FromString(s string) {

}

Or honestly, just a Parse(s string) method is probably more helpful.

Reply:
Hmmmm... I put in FromString and FromReader for flexibility.  Are you saying we should scrap FromReader